### PR TITLE
feat: upload lambda handler archive asset on release

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- update release workflow to zip up lambda handler files into an archive and upload as a release asset
- move test files into test directory to ensure they are not included in final archive
- update references to event and message locations in docs

## Motivation and Context
- the single `functions/notify_slack.py` file is getting long and cumbersome but its required in order to upload a single file. in order to start splitting up the functionality into different files, we need a way to provide users the lambda handler archive. this workflow is not user facing *yet, it just sets up the workflow of zipping the python files and uploading as a release asset. a later change will look at introducing the use of the archive similar to what is done [here](https://github.com/clowdhaus/terraform-aws-datadog-forwarders/blob/a404a62adb4133797066afc90cec3f5a5e4c803b/modules/log_forwarder/main.tf#L121)

## How Has This Been Tested?
- Currently testing on this fork https://github.com/clowdhaus/terraform-aws-notify-slack/releases; next step is to start using the zip archive but I need to re-check that the archive won't show up on the diff on every plan if there are no changes without introducing an S3 bucket. I think so, but TBD

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,6 +21,9 @@ jobs:
           persist-credentials: false
           fetch-depth: 0
 
+      - name: Create zip archive
```
Comment (antonbabenko) [inline .github/workflows/release.yml:24]: I think it is a good idea to keep the release process is identical across all our modules.

**Review by antonbabenko — COMMENTED** [review]
I am not sure I understand the purpose of making a zip and keeping it in assets. Do you want users to not depend on Python while using this module?

**Comment by bryantbiggs** [comment]
@antonbabenko I may have jumped the gun - I didn't know I could package up multiple files with the archive_file source. lets just ignore this PR for now

**Comment by antonbabenko** [comment]
Agree :) I still don't understand the reason for this pr. Have a great weekend!

**Comment by bryantbiggs** [comment]
I wanted to split up the single `notify_slack.py` file since it was getting long and it makes testing a bit more difficult, and adding events a bit harder. Originally I thought I needed to publish a zip file somewhere that contained all the python files needed (hence this PR), but instead I saw I can just use the archive_file data source. so nothing to see here 😅

**Pull request closed without merging**